### PR TITLE
feat: add identifiers compiler

### PR DESCRIPTION
**Pull request by IlyaFaer**
Adding an identifiers compiler class, which describes keywords and identifiers escaping.

### CODE REVIEW

**Comment by IlyaFaer** [comment]
This one is for cases like when a column was named as `asc` in reflection tests.

File: google/cloud/sqlalchemy_spanner/sqlalchemy_spanner.py [hunk 1]
```diff
@@ -131,12 +148,13 @@ def visit_drop_table(self, drop_table):
...
                 constrs += "ALTER TABLE {table} DROP CONSTRAINT {constr};".format(
-                    table=drop_table.element.name, constr=cons.name
+                    table=drop_table.element.name,
+                    constr=self.preparer.quote(cons.name),
                 )
 
         indexes = ""
         for index in drop_table.element.indexes:
-            indexes += "DROP INDEX {};".format(index.name)
+            indexes += "DROP INDEX {};".format(self.preparer.quote(index.name))
```
Comment (IlyaFaer) [inline google/cloud/sqlalchemy_spanner/sqlalchemy_spanner.py:157]: Quoting is needed here

File: test/test_suite.py [hunk 2]
```diff
@@ -632,12 +632,12 @@ def _test_get_unique_constraints(self, schema=None):
             # reserved identifiers
             Column("asc", sqlalchemy.String(30)),
             Column("key", sqlalchemy.String(30)),
+            sqlalchemy.Index("unique_a", "a", unique=True),
+            sqlalchemy.Index("unique_a_b_c", "a", "b", "c", unique=True),
+            sqlalchemy.Index("unique_c_a_b", "c", "a", "b", unique=True),
+            sqlalchemy.Index("unique_asc_key", "asc", "key", unique=True),
```
Comment (IlyaFaer) [inline test/test_suite.py:638]: Creating indexes right during table creation

**Comment by IlyaFaer** [comment]
Okay, `get_unique_constraints` test is now passing:

Безымянный

**Review by AVaksman — APPROVED** [review]

**Pull request merged**